tests/test_sensor.py

```diff
@@ -80,6 +82,49 @@ def status_func(value):
     assert sensor.status == Sensor.Status.NOMINAL
 
 
+def test_sensor_value_setter(diverse_sensors):
+    """Check whether the value type verification works."""
+    diverse_sensors["test-bool-sensor"].value = True
+    assert diverse_sensors["test-bool-sensor"]
+    with pytest.raises(TypeError):
+        diverse_sensors["test-bool-sensor"].value = "True"
+
+    diverse_sensors["test-int-sensor"].value = 1234
+    assert diverse_sensors["test-int-sensor"].value == 1234
+    with pytest.raises(TypeError):
+        diverse_sensors["test-int-sensor"].value = "1234"
+
+    diverse_sensors["test-float-sensor"].value = 1234.5
+    assert diverse_sensors["test-float-sensor"].value == 1234.5
+    with pytest.raises(TypeError):
+        diverse_sensors["test-float-sensor"].value = "1234"
+
+    diverse_sensors["test-str-sensor"].value = "foo"
+    assert diverse_sensors["test-str-sensor"].value == "foo"
+    with pytest.raises(TypeError):
+        diverse_sensors["test-str-sensor"].value = 1234
+
+    diverse_sensors["test-bytes-sensor"].value = b"bar"
+    assert diverse_sensors["test-bytes-sensor"].value == b"bar"
+    with pytest.raises(TypeError):
+        diverse_sensors["test-bytes-sensor"].value = "foo"
+
+    diverse_sensors["test-address-sensor"].value = "1.2.3.4"
+    assert diverse_sensors["test-address-sensor"].value == Address("1.2.3.4")
+    with pytest.raises(TypeError):
+        diverse_sensors["test-address-sensor"].value = 5678
+
+    diverse_sensors["test-timestamp-sensor"].value = 12345678
+    assert diverse_sensors["test-timestamp-sensor"].value == Timestamp(12345678)
+    with pytest.raises(TypeError):
+        diverse_sensors["test-timestamp-sensor"].value = "12345678"
+
+    diverse_sensors["test-enum-sensor"].value = MyEnum.ONE
+    assert diverse_sensors["test-enum-sensor"].value == MyEnum.ONE
+    with pytest.raises(TypeError):
+        diverse_sensors["test-enum-sensor"].value = OtherEnum.ABC
+
```

Comment:
We should also test with numpy scalars (will need to add numpy to requirements.in and project.optional-dependencies.test in pyproject.toml).